subiquity/models/kernel.py

```diff
@@ -40,11 +40,9 @@ def needed_kernel(self) -> Optional[str]:
         return self.metapkg_name
 
     def render(self):
+        result = {"kernel": {"remove_existing": True}}
         if self.curthooks_no_install:
-            return {"kernel": None}
-
-        return {
-            "kernel": {
-                "package": self.needed_kernel,
-            },
-        }
+            result["kernel"]["install"] = False
+        else:
+            result["kernel"]["package"] = self.needed_kernel
+        return result
```

Comment:
tangent: would be nice if we could use the rich curtin config types we've added here wouldn't it ...